### PR TITLE
Fix copy_transpose_valid check

### LINKED ISSUE #20755: Type conversion from float64 to float32 (cpu) sometimes crashes
```python
import torch

a = torch.rand(3, 3, dtype = torch.float64)
print(a.dtype, a.device) # torch.float64 cpu
c = a.to(torch.float32)
#works

b = torch.load('bug.pt')
print(b.dtype, b.device) # torch.float64 cpu
c = b.to(torch.float32)
# RuntimeError: expected scalar type Float but found Double

d = b.clone().to(torch.float32)
# works
```

[bug.zip](https://github.com/pytorch/pytorch/files/3202664/bug.zip)

### PR DESCRIPTION
**Pull request by colesbury**
Fixes #20755

(Was broken in #20685)

cc @vadimkantorov

### CODE REVIEW

**Comment by vadimkantorov** [comment]
:)

**Review by soumith — APPROVED** [review]

File: test/test_torch.py [hunk 1]
```diff
@@ -2644,6 +2644,18 @@ def test_copy_dtypes(self):
...
 
+    def test_copy_transpose(self):
+        x = torch.arange(100 * 100, dtype=torch.float).reshape(100, 100).t()
+        y = torch.empty(100, 100, dtype=torch.float)
+        y.copy_(x)
+        self.assertEqual(y[:, 0], range(100))
+        self.assertEqual(y[:, 40], range(4000, 4100))
+
+        y = torch.empty(100, 100, dtype=torch.double)
+        y.copy_(x)
```
Comment (vadimkantorov) [inline test/test_torch.py]: This tests copy from a transposed float array (x) to a contiguous float/double array (y). Though in my example it's double->float direction that breaks. Not sure it changes much though

Comment (colesbury) [reply test/test_torch.py]: The test captures the aspect that it's copying between different data types, which is all that matters here.

**Pull request closed without merging**